### ISSUE TITLE
`cast_abs_to_unsigned`

### ISSUE THREAD

**Issue by MrMuetze** [rust-lang/rust-clippy#8873]
### Summary

The lint `cast_abs_to_unsigned` is showing a warning even when the variable is cast to `usize`. The recommended code change does not produce working code and in the end leads to a more convoluted solution. Maybe the lint should not be triggered by code that is casting to `usize`.

### Lint Name

cast_abs_to_unsigned

### Reproducer

I tried this code:

```rust
fn main() {
    let a: i32 = -3;
    let _: usize = a.abs() as usize; // this produces the `cast_abs_to_unsigned` warning

    // let _: usize = a.unsigned_abs(); // fails with "mismatched types, expected `usize`, found `u32`"
    // let _: usize = a.unsigned_abs().try_into().unwrap(); // this is then recommended
}

```

I saw this happen:

```
warning: casting the result of `i32::abs()` to usize
 --> src/main.rs:3:20
  |
3 |     let _: usize = a.abs() as usize;
  |                    ^^^^^^^^^^^^^^^^ help: replace with: `a.unsigned_abs()`
  |
  = note: `#[warn(clippy::cast_abs_to_unsigned)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_abs_to_unsigned
```

I expected to see this happen: I suppose the lint does not apply to such a line of code.

### Version

```text
rustc 1.61.0 (fe5b13d68 2022-05-18)
```
